### ISSUE TITLE
No module named 'dataclasses' under Python 3.6

### ISSUE THREAD

**Issue by command-tab** [tiangolo/fastapi#997]
### Describe the bug

Under FastAPI 0.49.0 and Python 3.6, a simple API throws an exception:

```
File "./app/__init__.py", line 1, in <module>
--
  | from fastapi import FastAPI
  | File "/opt/app-root/lib/python3.6/site-packages/fastapi/__init__.py", line 7, in <module>
  | from .applications import FastAPI
  | File "/opt/app-root/lib/python3.6/site-packages/fastapi/applications.py", line 3, in <module>
  | from fastapi import routing
  | File "/opt/app-root/lib/python3.6/site-packages/fastapi/routing.py", line 6, in <module>
  | from fastapi.dependencies.models import Dependant
  | File "/opt/app-root/lib/python3.6/site-packages/fastapi/dependencies/models.py", line 3, in <module>
  | from fastapi.security.base import SecurityBase
  | File "/opt/app-root/lib/python3.6/site-packages/fastapi/security/__init__.py", line 2, in <module>
  | from .http import (
  | File "/opt/app-root/lib/python3.6/site-packages/fastapi/security/http.py", line 5, in <module>
  | from fastapi.exceptions import HTTPException
  | File "/opt/app-root/lib/python3.6/site-packages/fastapi/exceptions.py", line 3, in <module>
  | from fastapi.utils import PYDANTIC_1
  | File "/opt/app-root/lib/python3.6/site-packages/fastapi/utils.py", line 2, in <module>
  | from dataclasses import is_dataclass
  | ModuleNotFoundError: No module named 'dataclasses'
```

### To Reproduce

1. Create a simple API:

```
from fastapi import FastAPI

app = FastAPI(title='example', description='just a test', docs_url='/', redoc_url=None)

@app.get('/api/test')
async def index(foo: str = 'default', bar: str = None):
    return {
        'foo': foo,
        'bar': bar
    }
```

2. Attempt to run the API under Python 3.6 and observe the thrown exception

### Expected behavior

The API should be available and not raise an exception. However, it _does_ work under Python 3.7.

### Environment

- OS: Red Hat OpenShift, [Python 3.6 container](https://github.com/sclorg/s2i-python-container/tree/master/3.6)
- FastAPI Version: 0.49.0
- Python version: 3.6.3

**Comment by dmontagu** [comment]
The `dataclasses` backport to python 3.6 is a requirement for pydantic (at least for recent versions): https://github.com/samuelcolvin/pydantic/blob/645e5fe6a0c74c95c24410571e9bb804af3eb677/setup.py#L97

I'm assuming this is why its presence is expected.

**Comment by command-tab** [comment]
Interesting! I wonder why it didn't get installed when I installed FastAPI, and by extension, Pydantic 🤔

**Comment by dmontagu** [comment]
Yeah, if `pip install fastapi` doesn't result in the dataclasses package getting installed under python 3.6, I think we need to fix that.

**Comment by command-tab** [comment]
Ohhh, you know what, my local development environment is Python 3.7.x. I'll bet that `Pipfile.lock` doesn't include dataclasses because this environment didn't need it, but my OpenShift production environment needs it because it's 3.6. I bet if I were using a requirements.txt file instead of a full-graph lockfile, this wouldn't be an issue.

**Comment by dmontagu** [comment]
Makes sense! Let us know if you think there is any problematic behavior here that needs changing.

**Comment by command-tab** [comment]
I think we're all set here! I'll do one of:

- Adjust my development environment to 3.6.x so it matches my production (probably the best move)
- Use requirements.txt instead of a Pipfile/Pipfile.lock
- Wait for both environments to make Python 3.8 available and move everything to 3.8

Thanks for pointing me in the right direction 🙌

**Comment by tiangolo** [comment]
Thanks for the help here @dmontagu :cake: :bowing_man: 

Thanks @command-tab for reporting back and closing the issue :+1:
